libpod/container_internal.go

```diff
@@ -951,6 +952,33 @@ func (c *Container) generateSpec() (*spec.Spec, error) {
 		g.SetProcessGID(gid)
 	}
 
+	// Add addition groups if c.config.GroupAdd is not empty
+	if len(c.config.GroupAdd) > 0 {
+		if !c.state.Mounted {
+			return nil, errors.Wrapf(ErrCtrStateInvalid, "container %s must be mounted in order to add additional groups", c.ID())
+		}
+		for _, group := range c.config.GroupAdd {
+			gid64, err := strconv.ParseUint(group, 10, 32)
+			if err != nil {
+				_, gid32, err := chrootuser.GetUser(c.state.Mountpoint, strconv.Itoa(int(g.Spec().Process.User.UID))+":"+group)
+				if err != nil {
+					return nil, err
+				}
+				gid64 = uint64(gid32)
+			}
+			g.AddProcessAdditionalGid(uint32(gid64))
+		}
+	}
+
+	// Look up and add groups the user belongs to
+	groups, err := chrootuser.GetAdditionalGroupsForUser(c.state.Mountpoint, uint64(g.Spec().Process.User.UID))
```

Comment:
Do we need to do this? Won't this just be handled by /etc/group in the container?

Reply:
Well I think this is assigning the default groups from the container to the user that is joining.  Usually this is done via a login program, which we don't have here.

Reply:
Right, what @rhatdan said. Without this the default groups don't show up if we run the `id` command for example.